src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -733,7 +734,7 @@ CHIP_ERROR DefaultOTARequestor::SendQueryImageRequest(OperationalDeviceProxy & d
     ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSoftwareVersion(args.softwareVersion));
 
     args.protocolsSupported = kProtocolsSupported;
-    args.requestorCanConsent.SetValue(mOtaRequestorDriver->CanConsent());
+    args.requestorCanConsent.SetValue(!Basic::IsLocalConfigDisabled() && mOtaRequestorDriver->CanConsent());
```

Comment:
Can this be called in `ExtendedOTARequestorDriver`? In the default OTA requestor driver, it doesn't even make sense to check the attribute since the default driver cannot ask for the consent at all.

Reply:
We could, but what happens if someone implements another driver?  Per spec this should constrain _all_ driver impls, no matter what they decide about CanConsent internally.

Reply:
I think that implementing the driver requires quite some knowledge about the spec, anyway, but maybe you're right and we can do it here.